### PR TITLE
fix: Create item or class from subclasses from search result items in left sidebar (Search by label) for TAO Enterprise INVALSI PHP 8.1

### PR DESCRIPTION
**Pull request by chinnu-focaloid**
Related to https://oat-sa.atlassian.net/browse/LSI-1825

**Description**: Fix 500 error when clicking on 'New class'/'New Item' on search result items in left sidebar (Search by label) for TAO Enterprise INVALSI PHP 8.1

Whenever a user tries to click on 'New class'/'New Item' on search result items if the item is under subclasses, there’s a 500 error message popping up related to Security Exception and missing signature of the parent class. 

image

This issue can be reproduced in the following environment
http://test-invalsi_php_8_1.playground.kitchen.it.taocloud.org:49721/
admin/admin

To test this fix or enable the left sidebar search, you will need to activate the resourceSelector. In the config file config/tao/client_lib_config_registry.conf.php, simply add:

```
        'layout/tree/loader' => array(
            'treeProvider' => 'resource-selector'
        ),
```
The fix has been deployed to the following environment
http://test-chinnu.playground.kitchen.it.taocloud.org:41411/tao/Main/login
admin/admin

### CODE REVIEW

File: models/classes/resources/ListResourceLookup.php [hunk 1]
```diff
@@ -177,4 +179,13 @@ private function getSignatureGenerator(): SignatureGenerator
     {
         return $this->getServiceManager()->get(SignatureGenerator::class);
     }
+
+    private function getParentClass(core_kernel_classes_Resource $resource): ?core_kernel_classes_Class
+    {
+        $parentClassUri = $resource->getOnePropertyValue($resource->getProperty(OntologyRdf::RDF_TYPE));
+
```
Comment (wazelin) [inline models/classes/resources/ListResourceLookup.php:186]:
```suggestion

```

File: models/classes/resources/ListResourceLookup.php [hunk 2]
```diff
@@ -153,11 +152,14 @@ private function getResourceData($resource)
...
-                'label'      => $resource->getLabel(),
-                'type'       => 'instance',
-                'signature'  => $this->getSignatureGenerator()->generate($resource->getUri()),
+                'uri'               => $resource->getUri(),
+                'classUri'          => $resourceTypes[0],
+                'label'             => $resource->getLabel(),
+                'type'              => 'instance',
+                'signature'         => $this->getSignatureGenerator()->generate($resource->getUri()),
+                'classSignature'    => $this->getSignatureGenerator()->generate(
+                    $this->getParentClass($resource)->getUri()
```
Comment (wazelin) [inline models/classes/resources/ListResourceLookup.php:161]: `getParentClass()` return type is nullable, and this expression can result in an null-pointer exception.

**Review by wazelin — CHANGES_REQUESTED** [review]

**Review by wazelin — APPROVED** [review]

**Review by bartlomiejmarszal — COMMENTED** [review]
**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Review by bartlomiejmarszal — APPROVED** [review]
Great work!

**Pull request merged**